### ISSUE TITLE
Handle finally is not correct after update psalm to 3.18

### ISSUE THREAD

**Issue by bondarenko-vadym** [vimeo/psalm#4368]
https://psalm.dev/r/560c67d308
```php
   <?php

   function fooFunction (): string
   {
       try{
           $foo = 'foo';
       } finally {
       }
    
       return $foo;
   } 
```
Expected: No issues
Actual:
INFO: PossiblyUndefinedVariable - 10:12 - Possibly undefined variable $foo defined in try block

ERROR: InvalidReturnStatement - 10:12 - The inferred type 'string(foo)' does not match the declared return type 'string' for fooFunction

ERROR: InvalidReturnType - 3:26 - The declared return type 'string' for fooFunction is incorrect, got 'string(foo)'

**Comment by ADmad** [comment]
We are experiencing the same issue. No error should be reported in this case.

**Comment by weirdan** [comment]
psalm.dev also offers a fix but fails to change anything.

**Comment by muglug** [comment]
I pushed 3.18.1 and 4.0.1 to fix this – sorry folks.

I temporarily forgot how control flows out of `finally` blocks
